libraries/Microsoft.Bot.Builder.Ai.Translation/PostProcessor/PostProcessingUtilities.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Bot.Builder.Ai.Translation.PostProcessor
+{
+    internal class PostProcessingUtilities
+    {
+        /// <summary>
+        /// Helper to Join words to sentence
+        /// </summary>
+        /// <param name="delimiter">String delimiter used  to join words.</param> 
+        /// <param name="words">String Array of words to be joined.</param> 
+        /// <returns>string joined sentence</returns>
+        public static string Join(string delimiter, string[] words)
+        {
+            string sentence = string.Join(delimiter, words);
+            sentence = Regex.Replace(sentence, "[ ]?'[ ]?", "'");
+            return sentence.Trim();
+        }
+
+        /// <summary>
+        /// Helper to split sentence to words 
+        /// </summary>
+        /// <param name="sentence">String containing sentence to be splitted.</param> 
+        /// <returns>string array of words.</returns>
+        public static string[] SplitSentence(string sentence, string[] alignments = null, bool isSrcSentence = true)
+        {
+            string[] wrds = sentence.Split(' ');
+            string[] alignSplitWrds = new string[0];
+            if (alignments != null && alignments.Length > 0)
+            {
+                List<string> outWrds = new List<string>();
+                int wrdIndxInAlignment = 1;
+
+                if (isSrcSentence)
+                    wrdIndxInAlignment = 0;
+                else
+                {
+                    // reorder alignments in case of target translated  message to get ordered output words.
+                    Array.Sort(alignments, (x, y) => Int32.Parse(x.Split('-')[wrdIndxInAlignment].Split(':')[0]).CompareTo(Int32.Parse(y.Split('-')[wrdIndxInAlignment].Split(':')[0])));
+                }
+                string withoutSpaceSentence = sentence.Replace(" ", "");
+
+                foreach (string alignData in alignments)
```

Comment:
This is scary code, as I have no idea what it's doing and no means to judge if it's correct. 

Comments + Sampels, and lots of tests!!